### PR TITLE
Player rank stats

### PR DESCRIPTION
**Pull request by seattlecyclist**
This PR will add extended player rank stats.  Specifically it adds a queen_kills and other_kills.

Also some refactoring of the KillboardFull component such that the table building is a little more modular.

### CODE REVIEW

File: src/lib/GameStats.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import * as uuid from 'uuid/v4';
 import { KQStream, Character, PlayerKill } from './KQStream';
 
-type StatisticType = 'kills' | 'deaths';
+type StatisticType = 'kills' | 'queen_kills' | 'other_kills' | 'deaths';
```
Comment (seattlecyclist) [inline src/lib/GameStats.ts]: i ideally wanted to make the "kills" statistic be a rich object have nested attributes... but I punted... this is probably an area to improve.

Comment (KevinSnyderCodes) [reply src/lib/GameStats.ts]: Yea, something like "type of kill". Could indicate that a queen, warrior, or drone was killed.

File: src/client/Killboard.tsx [hunk 2]
```diff
@@ -74,76 +103,17 @@ export class KillboardFull extends Killboard {
...
-            <td>{this.state[Character.BlueSkulls].deaths}</td>
-          </tr>
-          <tr>
-            <td>
-              <img src={blueChecks} />
-            </td>
-            <td>{this.state[Character.BlueChecks].kills}</td>
-            <td>{this.state[Character.BlueChecks].deaths}</td>
-          </tr>
+          <KillboardRow {...{stat: this.state[Character.GoldQueen], image: goldQueen}}/>
```
Comment (KevinSnyderCodes) [inline src/client/Killboard.tsx]: Is this the same as this?

```ts
<KillboardRow stat={this.state[Character.GoldQueen]} image={goldQueen} />
```

Comment (seattlecyclist) [reply src/client/Killboard.tsx]: yes.. i can change that.  
I guess your way is the more standard react way so ill do it.

Comment (KevinSnyderCodes) [reply src/client/Killboard.tsx]: No worries, I'll take care of all the minor stuff like that :)

**Pull request merged**